src/CodeGen_LLVM.cpp

```diff
@@ -352,20 +366,34 @@ CodeGen_LLVM *CodeGen_LLVM::new_for_target(const Target &target,
                    << target.to_string() << "\n";
         return nullptr;
 
+#ifdef WITH_X86
     } else if (target.arch == Target::X86) {
         return make_codegen<CodeGen_X86>(target, context);
+#endif
+#ifdef WITH_ARM
     } else if (target.arch == Target::ARM) {
         return make_codegen<CodeGen_ARM>(target, context);
+#endif
+#ifdef WITH_MIPS
     } else if (target.arch == Target::MIPS) {
         return make_codegen<CodeGen_MIPS>(target, context);
+#endif
+#ifdef WITH_POWERPC
     } else if (target.arch == Target::POWERPC) {
         return make_codegen<CodeGen_PowerPC>(target, context);
+#endif
+#ifdef WITH_HEXAGON
     } else if (target.arch == Target::Hexagon) {
         return make_codegen<CodeGen_Hexagon>(target, context);
+#endif
+#ifdef WITH_WEBASSEMBLY
     } else if (target.arch == Target::WebAssembly) {
         return make_codegen<CodeGen_WebAssembly>(target, context);
+#endif
+#ifdef WITH_RISCV
     } else if (target.arch == Target::RISCV) {
         return make_codegen<CodeGen_RISCV>(target, context);
+#endif
     }
 
     user_error << "Unknown target architecture: "
```

Comment:
Again, we should provide a more informative error message. It's not _unknown_, it was deselected in the build configuration.